azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py

```diff
@@ -28,20 +28,20 @@ class MQTTTransportStage(PipelineStage):
     """
 
     @pipeline_thread.runs_on_pipeline_thread
-    def _cancel_active_connect_disconnect_ops(self):
+    def _cancel_pending_connection_op(self):
         """
-        Cancel any running connect or disconnect op.  Since our ability to "cancel" is fairly limited,
+        Cancel any running connect, disconnect or reconnect op. Since our ability to "cancel" is fairly limited,
         all this does (for now) is to fail the operation
         """
 
         ops_to_cancel = []
-        if self._active_connect_op:
+        if self._pending_connection_op:
             # TODO: should this actually run a cancel call on the op?
-            ops_to_cancel.add(self._active_connect_op)
-            self._active_connect_op = None
-        if self._active_disconnect_op:
-            ops_to_cancel.add(self._active_disconnect_op)
-            self._active_disconnect_op = None
+            ops_to_cancel.append(self._pending_connection_op)
+            self._pending_connection_op = None
+        if self._pending_connection_op:
+            ops_to_cancel.append(self._pending_connection_op)
+            self._pending_connection_op = None
 
         for op in ops_to_cancel:
```

Comment:
>   for op in ops_to_cancel: [](start = 5, length = 27)

There will only ever be one op in this list.  when you collapsed _active_connect_op and _active_disconnect_op, you removed the need to have a list here.

Reply:
That's what I thought. I'll remove this now.

---
In reply to: [311645622](https://github.com/Azure/azure-iot-sdk-python-preview/pull/159#discussion_r311645622) [](ancestors = 311645622)